### PR TITLE
[filters] fixing ignored `pcl::Indices` in `VoxelGrid` of `PCLPointCloud2`

### PR DESCRIPTION
**Pull request by kai-waang**
This PR fixed issue #5880. And it looks like there is a missing test in `VoxelGrid.Filters` which should test whether the given indices have been handled correctly.

### CODE REVIEW

**Comment by mvieth** [comment]
Hi, thank you for this pull request. It would actually be necessary to add two _new_ `getMinMax3D` functions, instead of adding the `indices` parameter to the existing functions. The reason is that people might use the existing `getMinMax3D` functions (without the `indices` parameter), and their code would no longer compile if we just add the parameter.

> And it looks like there is a missing test in `VoxelGrid.Filters` which should test whether the given indices have been handled correctly.

You are right, such a test would be good. If you like, feel free to add one, otherwise someone else might do that. An option could be to take the existing test, but add more points (e.g. very large points) to the cloud, and construct the indices vector such that `VoxelGrid` should disregard these additional points.

**Comment by kai-waang** [comment]
@mvieth I've added back these functions. I am not sure about how to add points to test. Should I add points to the pcd files or just  coding that in the test files.

**Comment by mvieth** [comment]
> @mvieth I've added back these functions. I am not sure about how to add points to test. Should I add points to the pcd files or just coding that in the test files.

No, please don't modify the PCD files. You could start by copying `cloud`, then adding points, then convert to `PCLPointCloud2` via `toPCLPointCloud2`. Something like:
```
PointCloud<PointXYZ>::Ptr cloud_modified (new PointCloud<PointXYZ>);
*cloud_modified = *cloud;
add points
PCLPointCloud2::Ptr cloud_blob_modified (new PCLPointCloud2);
toPCLPointCloud2(*cloud_modified, *cloud_blob_modified);
pass to voxel grid
```

**Comment by kai-waang** [comment]
@mvieth Thanks! I've added additional tests.

File: filters/src/voxel_grid.cpp [hunk 1]
```diff
@@ -174,6 +229,128 @@ pcl::getMinMax3D (const pcl::PCLPointCloud2ConstPtr &cloud, int x_idx, int y_idx
...
+             &cloud->data[xyz_offset[0] + index * cloud->point_step],
+             sizeof(float));
+      memcpy(&pt[1],
+             &cloud->data[xyz_offset[1] + index * cloud->point_step],
+             sizeof(float));
+      memcpy(&pt[2],
+             &cloud->data[xyz_offset[2] + index * cloud->point_step],
+             sizeof(float));
+      // Check if the point is invalid
+      if (!std::isfinite(pt[0]) || !std::isfinite(pt[1]) || !std::isfinite(pt[2])) {
```
[the remark this answers is not in the document]
Comment (kai-waang) [reply filters/src/voxel_grid.cpp:301]: Oh yes, I forgot to delete this check. Does this requires a further commit?

**Review by mvieth — APPROVED** [review]
Thank you!

**Review by larshg — APPROVED** [review]

**Pull request merged**